code/__DEFINES/traits.dm

```diff
@@ -114,6 +114,30 @@
 #define HAS_TRAIT_FROM_ONLY(target, trait, source) (target.status_traits?[trait] && (source in target.status_traits[trait]) && (length(target.status_traits[trait]) == 1))
 #define HAS_TRAIT_NOT_FROM(target, trait, source) (target.status_traits?[trait] && (length(target.status_traits[trait] - source) > 0))
 
+/// Define for adding multiple traits to a target via a list. Must have a common source and target.
+#define ADD_TRAIT_LIST(target, list_of_traits, source) \
+	do { \
+		if (!islist(list_of_traits)) { \
+			stack_trace("Invalid arguments passed to ADD_TRAIT_LIST! Invoked on [target] with [list_of_traits], source being [source]."); \
+			return; \
+		} \
+		for (var/trait in list_of_traits) { \
+			ADD_TRAIT(target, trait, source); \
+		} \
+	} while (0)
+
+/// Define for removing multiple traits from a target via a list. Must have a common source and target.
+#define REMOVE_TRAIT_LIST(target, list_of_traits, source) \
+	do { \
+		if (!islist(list_of_traits)) { \
+			stack_trace("Invalid arguments passed to REMOVE_TRAIT_LIST! Invoked on [target] with [list_of_traits], source being [source]."); \
+			return; \
+		} \
+		for (var/trait in list_of_traits) { \
+			REMOVE_TRAIT(target, trait, source); \
+		} \
+	} while (0)
```

Comment:
why are these while wrapped?

Reply:
i did it like that to mirror ADD_TRAIT and REMOVE_TRAIT being while scoped. i couldn't tell you why it's like that on a technical level, but I wanted to ensurethat we didn't really intefere with the operation of the base ADD/REMOVE_TRAIT function

i can run some tests to see if stuff breaks (maybe unit tests) if you think they don't need the while scope

Reply:
`ADD_TRAIT` etc are like that because they define a `var/list/_L`, and don't want to bleed it out. You aren't defining any variables so you wouldn't need this